tests/rptest/tests/partition_force_reconfiguration_test.py

```diff
@@ -204,3 +208,76 @@ def test_reconfiguring_with_dead_node(self, controller_snapshots):
         self.redpanda._admin.await_stable_leader(topic=self.topic,
                                                  replication=len(alive) + 1,
                                                  hosts=self._alive_nodes())
+
+    @cluster(num_nodes=7)
+    @matrix(target_replica_set_size=[1, 3])
+    def test_reconfiguring_all_replicas_lost(self, target_replica_set_size):
+        self.start_redpanda(num_nodes=4)
+        assert self.redpanda
+
+        # create a topic with rf = 1
+        self.topic = "topic"
+        self.client().create_topic(
+            TopicSpec(name=self.topic, replication_factor=1))
+
+        # produce some data.
+        self.start_producer(acks=1)
+        self.await_num_produced(min_records=10000)
+        self.producer.stop()
+
+        def get_stable_lso():
+            def get_lso():
+                state = self.redpanda._admin.get_partition_state(
+                    "kafka", self.topic, 0)
+                assert "replicas" in state.keys()
+                replicas = state["replicas"]
+                assert len(replicas) > 0
+                replica_lsos = [r["last_stable_offset"] for r in replicas]
+                # Check if all lsos are same
+                if all(lso == replica_lsos[0] for lso in replica_lsos):
+                    return replica_lsos[0]
+                return -1
+
+            wait_until(
+                lambda: get_lso() != -1,
+                timeout_sec=30,
+                backoff_sec=1,
+                err_msg=
+                f"Partition {self.topic}/0 couldn't achieve a stable lso")
+
+            return get_lso()
+
+        lso = get_stable_lso()
+        assert lso >= 10001, f"Partition {self.topic}/0 has incorrect lso {lso}"
+
+        # kill the broker hosting the replica
+        (killed, alive) = self._stop_majority_nodes(replication=1)
+        assert len(killed) == 1
+        assert len(alive) == 0
+
+        self._wait_until_no_leader()
+
+        # force reconfigure to target replica set size
+        assert target_replica_set_size <= len(self._alive_nodes())
+        new_replicas = [
+            Replica(dict(node_id=self.redpanda.node_id(replica), core=0)) for
+            replica in self.redpanda.started_nodes()[:target_replica_set_size]
+        ]
+        self._force_reconfiguration(new_replicas=new_replicas)
+
+        self.redpanda._admin.await_stable_leader(
+            topic=self.topic,
+            replication=target_replica_set_size,
+            hosts=self._alive_nodes())
+
+        # Ensure it is empty
+        lso = get_stable_lso()
+        # lso from debug partititon state is raft offset, hence 2 for an empty
+        # log with configuration batch.
+        assert lso == 2, f"Partition {self.topic}/0 has incorrect lso {lso}"
```

Comment:
can we use `>=` here, just in case there were more than one election

Reply:
or ideally check Kafka offset

Reply:
switched to kafka offset